### ISSUE TITLE
ValueError: width and height must be > 0

### ISSUE THREAD

**Issue by autumncp** [perrygeo/python-rasterstats#205]
Hello,

I am sorry if this is a simple question to answer, but I am getting an error message that says, what I am guessing, is something is wrong with the GeoTIFF I am using in my rasterstats analysis. I have ran this code before using other GeoTIFF files created from my 'tiffCreation.py' python script without error, but for some reason it is not working anymore. 

I have been looking at if for over two days now and I haven't found anything on the web that can help me. Can anyone tell me why this happens and how to fix it?

Files in the zip folder:
tiffCreation.py (where I created the GeoTIFF)
RasterStats.py (where I run rasterstats with my two files, this case acp.shp with any of the given GeoTIFFXX.tif files)
acp.shp
myGeoTIFF.tif (works)
myGeoTIFF_original.tif (does not work)
myGeoTIFF_200106c (does not work)

There are a decent amount of comments and commented code in these scripts, so I apologize if it's hard to follow. 

[stackoverflow.zip](https://github.com/perrygeo/python-rasterstats/files/4027931/stackoverflow.zip)

**Comment by autumncp** [comment]
So I have checked the width and height of the GeoTIFF that is leading to error and it is not zero in either width or height so I have not idea what the issue is. Still needing help here. 

Thanks

**Comment by perrygeo** [comment]
@autumncp For the GeoTIFF's that don't work, are they in the same projection? Your tif file needs to be in the same coordinate system as the `acp.shp` file.

**Comment by autumncp** [comment]
@perrygeo Yes! They're in the same projection. I have another geotiff that works and I have never changed the projection in any of the geotiffs I have created, its always EPSG 3857 for WGS84. Which is also what the acp.shp is in.

**Comment by autumncp** [comment]
So I have found the issue. It is with the resolution in geotransform. I somehow put a positive value for the resolution in the y-direction, but it needed to be negative. I am not sure how it got changed nor do I entirely understand why it needs to be negative for rasterstats zonal statistics. 
 The correction:
`geotransform = (xmin, xres, 0, ymax, 0, -yres)`
not
`geotransform = (xmin, xres, 0, ymin, 0, yres)`
 If anyone knows why this is and can explain it to me that would be very helpful!
Thanks!

**Comment by hoovco** [comment]
Been working on this for a couple days on my end ... for my experience, it looks like its a GDAL Grid issue. The output from grid results in a flipped image. If you reproject using:

warpOptions = gdal.WarpOptions(srcSRS='EPSG:4269', dstSRS='EPSG:4269') #whatever EPSG you need, make it the same
warp_out = gdal.Warp(rasterWarpOut, outputTest, options=warpOptions)

it corrects the image. Hope this helps others. Might be helpful in your case as well?
